//
// Copyright 2018 Asylo authors
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.
//

syntax = "proto2";

package asylo;

import "asylo/enclave.proto";

// Used to pass information in to the test shim enclave.
message TestShimEnclaveConfig {
  // GTest output file for detailed test results.
  optional string output_file = 1;
  // Location for all temporary test files.
  optional string test_tmpdir = 2;
  // Run tests in Initialize, rather than Run, if set and true.
  optional bool test_in_initialize = 3;
  // Regex that specifies the set of benchmarks to be executed.
  optional string benchmarks = 4;
}

extend EnclaveConfig {
  optional TestShimEnclaveConfig test_shim_enclave_config = 190514714;
}
